api/v1alpha1/hostedcluster_types.go

```diff
@@ -267,6 +269,51 @@ type ClusterAutoscaling struct {
 	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
 }
 
+// EtcdManagementType is a enum specifying the strategy for managing the clusters etcd instance
+type EtcdManagementType string
+
+const (
+	Managed   EtcdManagementType = "Managed"
+	Unmanaged EtcdManagementType = "Unmanaged"
+)
+
+type EtcdSpec struct {
+	// ManagementType defines how the etcd cluster is managed. Unmanaged means
+	// the etcd cluster is managed by a system outside the hypershift controllers.
+	// Managed means the hypershift controllers manage the provisioning of the etcd cluster
+	// and the operations around it
+	ManagementType EtcdManagementType `json:"managementType"`
```

Comment:
Needs union/enum/validation and defaulting metadata tags

Reply:
adding validation

Reply:
This field needs `// +unionDiscriminator`

Reply:
done